portable_concurrency/bits/promise.h

```diff
@@ -17,11 +17,15 @@ namespace detail {
 
 template<typename T>
 struct promise_common {
-  either<std::shared_ptr<shared_state<T>>, std::weak_ptr<shared_state<T>>> state_{
-    first_t{}, std::make_shared<shared_state<T>>()
-  };
-
-  promise_common() = default;
+  either<std::shared_ptr<shared_state<T>>, std::weak_ptr<shared_state<T>>> state_;
+
+  promise_common() :
+      state_{ first_t{}, std::make_shared<shared_state<T>>() }
+  { }
+  template<typename Alloc>
+  explicit promise_common(const Alloc& allocator) :
+      state_{ first_t{}, std::allocate_shared<shared_state<T>>(allocator) }
```

Comment:
This is good first step, but for complete allocator support state itself must become allocator aware. It allocates wait block on demand and allocates nodes for once_consumable_stack (which is simple forward-list with lock free pointer to head).